gcc/rust/expand/rust-macro-invoc-lexer.h

```diff
@@ -22,6 +22,33 @@
 #include "rust-ast.h"
 
 namespace Rust {
+template <class T> class MacroInvocLexerBase
+{
+public:
+  MacroInvocLexerBase (std::vector<T> stream)
```

Comment:
```suggestion
  MacroInvocLexerBase (std::vector<T> &&stream)
```

Reply:
https://stackoverflow.com/a/1849280

Reply:
But does the `MacroInvocLexer` need to store a copy? Or should it own its stream? I'm thinking it should own the vector, which might be created on the fly in the case of a `ProcMacroLexer` and thus needs to be moved to avoid unecessary copies